java/src/main/java/ai/onnxruntime/OrtUtil.java

```diff
@@ -493,6 +493,9 @@ static int capacityFromSize(int size) {
    * @return The prepared buffer tuple.
    */
   static BufferTuple prepareBuffer(Buffer data, OnnxJavaType type) {
+    if (type == OnnxJavaType.STRING || type == OnnxJavaType.UNKNOWN) {
+      throw new IllegalStateException("Cannot create a String tensor from a buffer, or an unknown tensor of any kind");
```

Comment:
>an unknown tensor of any kind [](http://example.com/codeflow?start=87&length=29)

Please, include the actual type in the message. The same below for the default clause.

Reply:
Fixed.